Refactor components, add TableOfContents and AiTaskMessage components, and update layout and navigation

guillermoscript opened the pull request:
This pull request includes several refactorings and enhancements to the codebase. It updates npm dependencies, refactors components to improve code structure and maintainability, and adds new components for displaying table of contents and AI task messages in lessons. It also updates layouts, navigation, and styling to enhance the user experience. Please review the changes and merge them into the main branch.

guillermoscript committed:
refactor: Update npm dependencies and refactor components

guillermoscript committed:
refactor: Update BreadcrumbComponent usage in multiple pages

The BreadcrumbComponent was refactored to be used consistently across multiple pages, including ChatPage, TeacherPage, CreateCoursePage, and CreateCourseLessonPage. This change simplifies the code structure, improves maintainability, and ensures a consistent user experience with the BreadcrumbComponent throughout the application.

guillermoscript committed:
refactor: Update LinkCourseCard component to use BreadcrumbComponent

The LinkCourseCard component was refactored to use the BreadcrumbComponent instead of the Breadcrumb, BreadcrumbItem, BreadcrumbLink, BreadcrumbList, and BreadcrumbSeparator components. This change simplifies the code structure and improves maintainability. It also ensures a consistent user experience with the BreadcrumbComponent across the application.

guillermoscript committed:
refactor: Add TableOfContents component for displaying lesson table of contents

A new component, TableOfContents, was added to display the table of contents for lessons. This component extracts headings from the lesson markdown and dynamically generates a list of links to each heading. It improves navigation within lessons and enhances the user experience. This change aligns with the refactoring done in other components, such as TeacherPage, CreateCoursePage, and EditLessonPage, which also enhance the lesson viewing experience.

guillermoscript committed:
refactor: Remove unused chat route files

The chat route files for the student and teacher have been removed as they are no longer needed. This cleanup improves the code structure and reduces unnecessary files in the repository.

guillermoscript committed:
```text
refactor: Add smooth scrolling behavior to HTML element

This commit adds the `scroll-behavior: smooth;` CSS property to the HTML element in the `globals.css` file. This change enables smooth scrolling behavior when navigating within the application. It enhances the user experience by providing a more visually pleasing and seamless scrolling experience.

guillermoscript committed:
refactor: Add studentSubmitAiTaskMessage function for submitting AI task messages

The `studentSubmitAiTaskMessage` function is added to the `lessonsAction.ts` file. This function allows students to submit AI task messages for a specific lesson. It retrieves the user data, inserts the message into the database, and returns a success response. This change enhances the functionality of the application by enabling students to communicate with the AI assistant for their lessons.

guillermoscript committed:
```text
refactor: Update session middleware to handle user role redirection

The session middleware in the `middleware.ts` file has been updated to handle user role redirection. This change ensures that users are redirected to the appropriate dashboard based on their role (teacher, student, or admin). It improves the user experience by providing personalized access to the dashboard and aligns with the refactoring done in other components.

guillermoscript committed:
refactor: Extract headings from lesson markdown to generate table of contents

This commit adds a new function, `extractHeadings`, to the `functions.ts` file. The function extracts headings from the lesson markdown and dynamically generates a table of contents. It improves navigation within lessons and enhances the user experience. This change aligns with the refactoring done in other components, such as TeacherPage, CreateCoursePage, and EditLessonPage, which also enhance the lesson viewing experience.

guillermoscript committed:
refactor: Update ViewMarkdown component to improve rendering and navigation

This commit updates the ViewMarkdown component in the ViewMarkdown.tsx file. It adds new dependencies, such as 'github-markdown-css', 'rehype-autolink-headings', 'rehype-slug', and 'remark-rehype', to enhance the rendering and navigation of markdown content. The changes include using the 'react-markdown' library, 'Prism' syntax highlighter, and various plugins for remark and rehype. This refactor improves the readability and user experience of markdown content in the application.

guillermoscript committed:
refactor: Update Timeline component to improve readability and user experience

This commit refactors the Timeline component in the timeline.tsx file. It simplifies the code structure by removing unnecessary forwardRef usage and consolidating related components. The changes include extracting the TimelineItem, TimelineTime, TimelineConnector, TimelineHeader, TimelineTitle, TimelineIcon, TimelineDescription, and TimelineContent components into separate files for better organization. This refactor improves the readability and user experience of the Timeline component in the application.

guillermoscript committed:
refactor: Improve Timeline component readability and user experience

guillermoscript committed:
refactor: Update LessonsTimeLine component to use ol element and separate TimelineItem into its own component

This commit refactors the LessonsTimeLine component in the LessonsTimeLine.tsx file. It replaces the Timeline component with an ol element for better semantic structure. It also extracts the TimelineItem component into its own file for improved organization. These changes improve the readability and user experience of the LessonsTimeLine component in the application.

guillermoscript committed:
refactor: Update LessonPage layout to include a border on the sidebar

This commit updates the LessonPage component in the LessonPage.tsx file. It adds a border to the sidebar section of the layout, improving the visual separation between the main content and the sidebar. This change enhances the overall design and user experience of the LessonPage component.

guillermoscript committed:
refactor: Improve LessonNavigationButtons component layout and accessibility

This commit improves the LessonNavigationButtons component in the LessonNavigationButtons.tsx file. It updates the layout by adding a flexbox container and applying responsive classes for better alignment and spacing. Additionally, it adds an aria-label attribute to improve accessibility by providing a descriptive label for screen readers. These changes enhance the overall user experience and make the component more accessible.

guillermoscript committed:
refactor: Remove unused import in LessonComplete.tsx

guillermoscript committed:
feat: Add AiTaskMessage component for displaying AI task messages in lessons

This commit adds the AiTaskMessage component to the lessons directory. The component is responsible for rendering AI task messages in lessons and includes logic for retrieving and displaying messages from the database. It enhances the functionality of the application by allowing students to interact with the AI assistant during their lessons.

guillermoscript committed:
refactor: Improve LessonNavigationButtons component layout and accessibility

guillermoscript committed:
refactor: Remove unused imports and update CourseCard component

This commit removes the unused imports in the CourseCard component and updates its structure. The unnecessary imports of lucide-react, buttonVariants, and cn are removed. Additionally, the View Lessons and View Exams buttons are removed from the CourseCard component. These changes improve the code cleanliness and remove unnecessary dependencies in the component.

guillermoscript committed:
refactor: Update AllCoursesCard component to use LinkCourseCard instead of EnrollButton

guillermoscript committed:
refactor: Update CommentCard component to improve reaction button layout and functionality

guillermoscript committed:
refactor: Use client-side rendering in chat component

guillermoscript committed:
refactor: Update CommentsSections component layout and styling

This commit updates the CommentsSections component in the CommentsSections.tsx file. It improves the layout and styling by adding padding to the CardHeader and CardContent elements. Additionally, it adds a responsive class to the CardHeader for better alignment on larger screens. These changes enhance the visual presentation and user experience of the CommentsSections component.

guillermoscript committed:
refactor: Update DashboardHeader component layout for responsiveness

guillermoscript committed:
refactor: Update Header component to include Lucide icons and improve navigation

This commit updates the Header component in the Header.tsx file. It adds the Lucide icons for DollarSign and ShoppingBag to enhance the visual presentation of the navigation menu. Additionally, it improves the navigation by adding links to the Plans and Store pages. These changes enhance the overall user experience and make the navigation more intuitive.

guillermoscript committed:
refactor: Mobile view fix also total refacto on the AI part and added new components for the timeline and table of content

guillermoscript committed:
refactor: Update StudentExamCoursePage to display teacher's full name

This commit updates the StudentExamCoursePage component to display the full name of the teacher who created the exam. It fetches the teacher's data from the 'profiles' table in the database and renders it in the UI. This enhancement provides better clarity and information to the students taking the exam.

guillermoscript committed:
refactor: Update chat link label to truncate long titles

This commit updates the chat link label in the ExamnChatIdPageLayout component to truncate long titles. It slices the title to a maximum of 40 characters and appends '...' to indicate truncation. This change improves the visual presentation of the chat link and prevents it from overflowing in the UI.

guillermoscript committed:
refactor: Redirect to dashboard if user is already logged in

This commit adds a redirect to the dashboard if the user is already logged in. It checks if the request URL starts with '/auth' and if the user data is available. If both conditions are met, it redirects to the dashboard based on the user's role. This enhancement improves the user experience by automatically redirecting logged-in users to the appropriate dashboard.

guillermoscript committed:
refactor: Add BreadcrumbSeparator component to Breadcrumb module

guillermoscript committed:
refactor: Update BreadcrumbComponent to use next/link for breadcrumb links

guillermoscript committed:
refactor: Truncate chat link label to 40 characters

This commit updates the ExamnChatIdPageLayout component to truncate the chat link label to a maximum of 40 characters. It improves the visual presentation of the chat link and prevents it from overflowing in the UI.

angel-afonso commented on actions/dashboard/AI/TaskAiActions.tsx:
```diff
@@ -0,0 +1,263 @@
+
+    const supabase = createClient()
+
+    console.log(input)
```
remove this

angel-afonso commented on actions/dashboard/AI/TaskAiActions.tsx:
```diff
@@ -0,0 +1,263 @@
+        (message) => message.role === 'system'
+    )
+
+    console.log(systemMessage)
```
remove this

angel-afonso commented on actions/dashboard/AI/TaskAiActions.tsx:
```diff
@@ -0,0 +1,263 @@
+                    created_at: new Date().toISOString()
+                })
+
+                console.log(aiMessageInsert)
```
Maybe this log is unnecesary

angel-afonso commented on actions/dashboard/AI/TaskAiActions.tsx:
```diff
@@ -0,0 +1,263 @@
+                        user_id: aiState.get().userId
+                    })
+
+                    console.log(task)
```
Remove this

angel-afonso commented on actions/dashboard/lessonsAction.ts:
```diff
@@ -134,3 +134,36 @@ export async function deleteLessonsAction (data: {
+            console.log('Error adding message to the database', messageData.error)
+            return createResponse('error', 'Error adding message to the database', null, 'Error adding message to the database')
+        }
+        console.log('Message added to the database', messageData)
```
Remove this

angel-afonso commented on components/dashboards/student/course/lessons/TaksMessages.tsx:
```diff
@@ -1,125 +1,133 @@
+                                },
+                            })
+
+                            console.log(res)
```
remove this

angel-afonso reviewed (changes requested):
Remove the unnecesary console.log

guillermoscript committed:
refactor: Update package.json and package-lock.json dependencies

This commit updates the dependencies in the package.json and package-lock.json files. It adds the "framer-motion" package with version "^11.3.8" and the "mini-svg-data-uri" package with version "^1.4.4". These updates ensure that the project uses the latest stable versions of these dependencies, which may introduce new features, bug fixes, and performance improvements.

guillermoscript committed:
feat: Add DotBackground and GridBackground components

This commit adds the DotBackground and GridBackground components to the Acernity UI module. These components provide a background with dot and grid patterns respectively, enhancing the visual appeal of the UI. They are implemented using CSS classes and utilize framer-motion for animations. This addition expands the range of available background options for developers to use in their projects.

guillermoscript committed:
refactor: Update LessonPage sidebar styling

This commit updates the LessonPage component's sidebar styling in the page.tsx file. The className for the Tabs component is modified to include "flex items-center justify-center flex-col", improving the alignment and layout of the sidebar. This change enhances the visual presentation and user experience of the lesson page.

guillermoscript committed:
refactor: Update package.json and package-lock.json dependencies

guillermoscript committed:
refactor: Update package.json and package-lock.json dependencies

guillermoscript committed:
feat: Add Confetti component to MagicUI module

This commit adds the Confetti component to the MagicUI module. The Confetti component is a reusable React component that creates a confetti effect using the canvas-confetti library. It provides an API for firing confetti with customizable options. This addition enhances the visual appeal and interactivity of the UI, allowing developers to easily incorporate confetti effects into their applications.

guillermoscript committed:
refactor: Add fire prop to ExamPreparationActions component

This commit adds the `fire` prop to the `ExamPreparationActions` component in the `ExamPreparationActions.tsx` file. The `fire` prop is missing a value and needs to be updated accordingly. This change ensures that the component functions correctly and avoids any potential errors or unexpected behavior.

guillermoscript committed:
refactor: Remove console.log statements from TaskAiActions.tsx

guillermoscript committed:
feat: Add ReviewCard component to dashboards/cards

This commit adds the ReviewCard component to the dashboards/cards directory. The ReviewCard component is a reusable React component that displays a review with a star rating, review text, reviewer name, and review date. It enhances the functionality and visual presentation of the dashboard cards, allowing developers to easily incorporate review cards into their applications.

guillermoscript committed:
feat: Add ListOfReviews component to dashboards/Common/reviews

This commit adds the ListOfReviews component to the dashboards/Common/reviews directory. The ListOfReviews component is a reusable async function that fetches and displays a list of reviews based on the provided entityId and entityType. It utilizes the Supabase library to query the 'reviews' table and renders ReviewCard components for each review. This addition enhances the functionality and visual presentation of the reviews section in the dashboard, allowing developers to easily incorporate review lists into their applications.

guillermoscript committed:
feat: Add ReviewForm component to dashboards/Common/reviews

This commit adds the ReviewForm component to the dashboards/Common/reviews directory. The ReviewForm component is a form component that allows users to submit reviews with a rating and review text. It utilizes the react-hook-form library for form validation and submission. This addition enhances the functionality of the reviews section in the dashboard, allowing users to easily submit reviews and provide feedback.

guillermoscript committed:
feat: Add ConfettiMessage component to dashboards/student/chat/exam-prep

This commit adds the ConfettiMessage component to the dashboards/student/chat/exam-prep directory. The ConfettiMessage component is a React component that displays a confetti effect using the Confetti component from the MagicUI module. It utilizes the useRef and useEffect hooks to fire the confetti effect when the component mounts. This addition enhances the visual appeal and interactivity of the chat interface, allowing developers to easily incorporate confetti messages into their applications.

guillermoscript committed:
feat: Add ParticlesSection component to home page

This commit adds the ParticlesSection component to the home page. The ParticlesSection component is a React component that displays a particle effect using the Particles component from the MagicUI module. It dynamically sets the color of the particles based on the theme selected using the useTheme hook from the next-themes library. This addition enhances the visual appeal and interactivity of the home page, allowing developers to easily incorporate particle effects into their applications.

guillermoscript committed:
feat: Add task_instructions field to Database type

This commit adds the `task_instructions` field to the `Database` type in the `supabase.ts` file. The `task_instructions` field is added to the `Row`, `Insert`, and `Update` types within the `Database` type. This change allows for the storage and manipulation of task instructions in the database, enhancing the functionality of the application.

guillermoscript committed:
feat: Add lazy loading for images in ViewMarkdown component

guillermoscript committed:
feat: Add container class to Header component for improved styling

guillermoscript committed:
feat: Add addReview function to studentActions.ts

This commit adds the addReview function to the studentActions.ts file. The addReview function allows users to add a review with a star rating and review text. It utilizes the Supabase library to insert the review into the 'reviews' table in the database. This addition enhances the functionality of the application, allowing users to easily submit reviews and provide feedback.

guillermoscript committed:
feat: Add FeaturedSection component to home page

This commit adds the FeaturedSection component to the home page. The FeaturedSection component is a React component that displays a grid of features with icons, titles, and descriptions. It enhances the visual presentation and functionality of the home page, allowing developers to easily showcase key features of their application.

guillermoscript committed:
feat: Add Guided Tutorial button to ChatTourFreeChat component

guillermoscript committed:
feat: Add Guided Tutorial button to ChatTourExamPrepChat component

guillermoscript committed:
feat: Add ConfettiMessage component to ExamFeedbackCard

This commit adds the ConfettiMessage component to the ExamFeedbackCard in the dashboards/student/chat directory. The ConfettiMessage component is a React component that displays a confetti effect using the Confetti component from the MagicUI module. It enhances the visual appeal and interactivity of the chat interface, allowing developers to easily incorporate confetti messages into their applications.

guillermoscript committed:
feat: Add lazy loading for images in ReviewForm component

guillermoscript committed:
feat: Add ReviewForm component to ListOfReviews

This commit adds the ReviewForm component to the ListOfReviews file in the components/dashboards/Common/reviews directory. The ReviewForm component allows users to submit reviews with star ratings and review text. It enhances the functionality of the ListOfReviews component, allowing users to easily add their own reviews.

guillermoscript committed:
feat: Add relative positioning to Message component

This commit adds the 'relative' class to the Message component in the components/dashboards/Common/chat/Message.tsx file. The 'relative' class is added to the className prop of the div element, enhancing the layout and positioning of the Message component in the chat interface.

guillermoscript committed:
feat: Add Confetti component to SuccessMessage

This commit adds the Confetti component to the SuccessMessage component in the components/dashboards/Common/chat/chat.tsx file. The Confetti component is imported from the '@/components/magicui/confetti' module and is rendered when the 'fire' prop is true. It enhances the visual appeal and interactivity of the SuccessMessage component, allowing developers to easily incorporate confetti effects into their chat interface.

guillermoscript committed:
major refactor of the lessons page to add now reviews, links to markdown as optional parma set, added instructions for the end user of what the Task is required, ui changes

guillermoscript committed:
feat: Add lazy loading for images in ReviewForm component

guillermoscript committed:
build fix

guillermoscript committed:
feat: Remove console.log statements and update pricing section on PlanPage

The commit removes unnecessary console.log statements from the TaskAiActions and lessonsAction files. It also updates the pricing section on the PlanPage by importing the PricingSection component and making layout adjustments.

guillermoscript committed:
feat: Add TierCard component

This commit adds the TierCard component to the components/plans directory. The TierCard component is a reusable React component that displays information about a pricing tier, including the title, price, description, and features. It enhances the functionality of the plans page, allowing users to easily compare and select different pricing options.
